package/kedro_viz/integrations/kedro/lite_parser.py

```diff
@@ -0,0 +1,242 @@
+import ast
+import logging
+from collections import defaultdict
+from pathlib import Path
+from typing import Dict, Iterable, List
+
+from kedro.pipeline.modular_pipeline import pipeline as ModularPipeline
+from kedro.pipeline.pipeline import Node, Pipeline
+
+logger = logging.getLogger(__name__)
+
+
+class KedroPipelineLocator(ast.NodeVisitor):
+    """
+    Represents a pipeline that is located when parsing
+    the Kedro project's `create_pipeline` function
+
+    """
+
+    def __init__(self):
+        self.pipeline = None
+
+    def visit_FunctionDef(self, node):
+        try:
+            if node.name == "create_pipeline":
+                # Explore the located pipeline for nodes
+                # and other keyword args
+                kedro_pipeline_explorer = KedroPipelineExplorer()
+                kedro_pipeline_explorer.visit(node)
+                try:
+                    # modular pipeline
+                    if kedro_pipeline_explorer.namespace:
+                        self.pipeline = ModularPipeline(
+                            pipe=kedro_pipeline_explorer.nodes,
+                            inputs=kedro_pipeline_explorer.inputs,
+                            outputs=kedro_pipeline_explorer.outputs,
+                            parameters=kedro_pipeline_explorer.parameters,
+                            tags=kedro_pipeline_explorer.tags,
+                            namespace=kedro_pipeline_explorer.namespace,
+                        )
+                    else:
+                        # kedro pipeline
+                        self.pipeline = Pipeline(
+                            nodes=kedro_pipeline_explorer.nodes,
+                            tags=kedro_pipeline_explorer.tags,
+                        )
+                except Exception as exc:
+                    # [TODO: Error with modular pipeline, try creating regular pipeline]
+                    logger.error(exc)
+                    self.pipeline = Pipeline(
+                        nodes=kedro_pipeline_explorer.nodes,
+                        tags=kedro_pipeline_explorer.tags,
+                    )
+
+            self.generic_visit(node)
+
+        except Exception as exc:
+            # [TODO: Error with parsing the file,
+            # dump the visiting node for debugging]
+            logger.error(exc)
+            logger.info(ast.dump(node, indent=2))
+
+
+class KedroPipelineExplorer(ast.NodeVisitor):
+    # [TODO: Current explorer only serves for 1 pipeline() function within a create_pipeline def]
+    def __init__(self):
+        # keeping these here for future use-case
+        # when dealing with multiple `pipeline()` functions
```

Comment:
As you cannot use `find_pipeline`, you will likely need something like this to discover the pipeline in a similar way. 

https://github.com/kedro-org/vscode-kedro/blob/91e14d320e0d94b5bc3dc05148955f0f16552308/bundled/tool/lsp_server.py#L482-L499